src/main/java/com/iexec/core/task/TaskRepository.java

```diff
@@ -44,6 +44,21 @@
     @Query("{ 'currentStatus': {$in: ?0} }")
     List<Task> findByCurrentStatus(List<TaskStatus> statuses, Sort sort);
 
+    @Query("{ 'currentStatus': {$in: ?0} , 'tag': {$ne: ?1} }")
+    List<Task> findByCurrentStatusInAndTagNot(List<TaskStatus> statuses, String excludedTag, Sort sort);
+
+    // Shortcut for `findFirstByCurrentStatusInAndTagNotAndChainTaskIdNotIn`
+    default Optional<Task> findFirstTask(List<TaskStatus> statuses, String excludedTag, List<String> excludedChainTaskIds, Sort sort) {
```

Comment:
`findOneLikelyAvailableTask`
?
Do we need default?

Reply:
I still prefer to keep `first` in the method name as results are sorted before the first one is returned. BTW, I also prefer to stay generic as this does not simply find available tasks, it finds tasks according to criteria. Maybe something like `findFirstTaskGivenCritiria`?

We need `default` as this is a method whose body is defined in an interface.

Reply:
If you really want to add a alias on this method (`findFirstTaskGiven`) I suggest you attach a full javadoc

Reply:
I really think it's better to have an alias, yes. The other method name is really, really long so providing an alias is mandatory IMHO. I'll add some JavaDoc to explain in-depth what's exactly this alias.

Reply:
Regarding the naming: what about using `prioritized` there, too? Something like `findPrioritizedTask`?

Reply:
See https://github.com/iExecBlockchainComputing/iexec-core/pull/490/commits/e02f7a6bca5fefefe56b1b901f9c5fdc0418d7bc.